helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -241,38 +241,28 @@ private boolean computeResourceBestPossibleState(ClusterEvent event, ResourceCon
       try {
         HelixManager manager = event.getAttribute(AttributeName.helixmanager.name());
         rebalancer.init(manager);
-        idealState =
-            rebalancer.computeNewIdealState(resourceName, idealState, currentStateOutput, cache);
+        idealState = rebalancer.computeNewIdealState(resourceName, idealState, currentStateOutput, cache);
 
         output.setPreferenceLists(resourceName, idealState.getPreferenceLists());
 
         // Use the internal MappingCalculator interface to compute the final assignment
         // The next release will support rebalancers that compute the mapping from start to finish
-        partitionStateAssignment = mappingCalculator
-            .computeBestPossiblePartitionState(cache, idealState, resource, currentStateOutput);
+        partitionStateAssignment =
+            mappingCalculator.computeBestPossiblePartitionState(cache, idealState, resource, currentStateOutput);
         for (Partition partition : resource.getPartitions()) {
           Map<String, String> newStateMap = partitionStateAssignment.getReplicaMap(partition);
           output.setState(resourceName, partition, newStateMap);
         }
 
         // Check if calculation is done successfully
         return checkBestPossibleStateCalculation(idealState);
+      } catch (HelixException e){
+        // No eligible instance is found.
+        LogUtil
+            .logError(logger, _eventId, e.getMessage());
       } catch (Exception e) {
         LogUtil
             .logError(logger, _eventId, "Error computing assignment for resource " + resourceName + ". Skipping.", e);
-        // TODO : remove this part after debugging NPE
-        StringBuilder sb = new StringBuilder();
```

Comment:
@dasahcc: Can you confirm this part can be safely removed?

Reply:
@lei-xia Yes 3 commits are for one issue. I will combine them into one commit and create another pull request. Other one is just for deleting the logs. I can create a separate issue for that. I already confirmed this log removal part with Junkai. @dasahcc can you please have a look and confirm it again.

Thanks.

Reply:
Yes. They can be removed since the NPE happened by assignment is NULL. If we check that part, we should fine with the logs.

But you have too many commits. I would suggest to merge some of them into to commits: CRUSHED log fixing and removing the todo log. Put them into two separated PRs.